src/App.tsx

```diff
@@ -1,107 +1,114 @@
-import React from 'react';
-import flag from './pngegg.png'
-import logo from './logo.svg'
-
-import './App.css';
+import React from "react";
+import flag from "./pngegg.png";
+import "./App.css";
 
 function App() {
-  const iniImg = {1:flag, 2:flag, 3:flag, 4:flag,
-    5:flag, 6:flag, 7:flag, 8:flag, 9:flag, 10:flag, 11:flag, 12:flag, 13:flag, 14:flag, 15:flag,
-    16:flag, 17:flag, 18:flag, 19:flag, 20:flag, 21:flag};
-  const [img, setImg] = React.useState(iniImg);
+  const iniImg = new Array(21).fill(flag);
   const [flags, setFlags] = React.useState(21);
-  const [player, setPlayer] = React.useState('first');
+  const [player, setPlayer] = React.useState("first");
   const [took, setTook] = React.useState(0);
-  const [previousPlayer, setpreviousPlayer] = React.useState('first');
-
+  const [previousPlayer, setpreviousPlayer] = React.useState("first");
 
-  const onclick1 = () => {
-    setFlags(flags - 1);
-    setImg(prevState => ({
-      ...prevState,
-      [22-flags]: logo,
-    }))
-    if (flags === 1 || flags === 0) {
-      alert('Player ' + player + ' won' );
+  // it maps trough filtered array to make rows
+  // 'a' and 'b' changes the shape
+  const returnFlag = (a: number, b: number) => {
+    return (
+      <div key={"a"}>
```

Comment:
I think key is not needed here - https://reactjs.org/docs/lists-and-keys.html#keys

You should use unique keys for elements that are inside of an array